adapter/internal/oasparser/envoyconf/routes_with_clusters.go

```diff
@@ -94,26 +95,36 @@ func CreateRoutesWithClusters(mgwSwagger model.MgwSwagger, interceptorCerts map[
 	clusters = append(clusters, clustersI...)
 	endpoints = append(endpoints, endpointsI...)
 
+	// Maintain a clusterName-EndpointCluster mapping to prevent duplicate
+	// creation of clusters.
+	processedEndpoints := map[string]model.EndpointCluster{}
+
 	for _, resource := range mgwSwagger.GetResources() {
+		var clusterName string
 		resourcePath := resource.GetPath()
 		endpoint := resource.GetEndpoints()
 		basePath := strings.TrimSuffix(endpoint.Endpoints[0].Basepath, "/")
-		clusterName := getClusterName(endpoint.EndpointPrefix, organizationID, vHost, mgwSwagger.GetTitle(), apiVersion, resource.GetID())
-		cluster, address, err := processEndpoints(clusterName, endpoint, timeout, basePath)
-		if err != nil {
-			logger.LoggerOasparser.Errorf("Error while adding resource level endpoints for %s:%v-%v. %v",
-				apiTitle, apiVersion, resourcePath, err.Error())
+		existingClusterName := getExistingClusterName(*endpoint, processedEndpoints)
+		
+		if existingClusterName == "" {
+			clusterName = getClusterName(endpoint.EndpointPrefix, organizationID, vHost, mgwSwagger.GetTitle(), apiVersion, resource.GetID())
+			cluster, address, err := processEndpoints(clusterName, endpoint, timeout, basePath)
+			if err != nil {
+				logger.LoggerOasparser.Errorf("Error while adding resource level endpoints for %s:%v-%v. %v",
```

Comment:
please follow the new error log format. example:
```
logger.LoggerXds.ErrorC(logging.GetErrorByCode(2231, mgwSwagger.GetTitle(), mgwSwagger.GetVersion(), resource.GetPath(), err.Error()))
```

Reply:
fixed